src/acceptance-test/java/org/zalando/nakadi/repository/db/SubscriptionDbRepositoryTest.java

```diff
@@ -108,26 +111,65 @@ public void whenListSubscriptionsThenOk() throws ServiceUnavailableException {
+                randomSubscription().withOwningApplication("app").withEventType("et1").build(),
+                randomSubscription().withOwningApplication("app").withEventType("et2").build(),
+                randomSubscription().withOwningApplication("app").withEventTypes(ImmutableSet.of("et2", "et3")).build(),
+                randomSubscription().withOwningApplication("app2").withEventType("et1").build(),
+                randomSubscription().withOwningApplication("app2").withEventType("et2").build());
+        testSubscriptions.forEach(this::insertSubscriptionToDB);
+
+        final List<Subscription> expectedSubscriptions = testSubscriptions.stream()
+                .filter(sub -> "app".equals(sub.getOwningApplication()) && sub.getEventTypes().contains("et1"))
+                .sorted(SUBSCRIPTION_CREATION_DATE_DESC_COMPARATOR)
+                .collect(toList());
+
+        final List<Subscription> subscriptions = repository.listSubscriptions(ImmutableSet.of("et1"),
+                Optional.of("app"), 0, 10);
+        assertThat(subscriptions, equalTo(expectedSubscriptions));
+    }
+
+    @Test
+    public void whenListSubscriptionsByMultipleEventTypesThenOk() throws ServiceUnavailableException {
+        final List<Subscription> testSubscriptions = ImmutableList.of(
+                randomSubscription().withEventTypes(ImmutableSet.of("et1", "et2")).build(),
+                randomSubscription().withEventTypes(ImmutableSet.of("et1", "et2", "et3")).build(),
+                randomSubscription().withEventTypes(ImmutableSet.of("et1")).build(),
+                randomSubscription().withEventTypes(ImmutableSet.of("et2")).build(),
+                randomSubscription().withEventTypes(ImmutableSet.of("et3", "et4", "et5")).build());
+        testSubscriptions.forEach(this::insertSubscriptionToDB);
+
+        final List<Subscription> expectedSubscriptions = testSubscriptions.stream()
+                .filter(sub -> sub.getEventTypes().containsAll(ImmutableSet.of("et1", "et2")))
+                .sorted(SUBSCRIPTION_CREATION_DATE_DESC_COMPARATOR)
+                .collect(toList());
 
+        final List<Subscription> subscriptions = repository.listSubscriptions(ImmutableSet.of("et1", "et2"),
+                Optional.empty(), 0, 10);
+        assertThat(subscriptions, equalTo(expectedSubscriptions));
+    }
+
+    @Test
+    public void whenListSubscriptionsLimitAndOffsetAreRespected() throws ServiceUnavailableException {
+        final List<Subscription> testSubscriptions = createRandomSubscriptions(10);
         testSubscriptions.forEach(this::insertSubscriptionToDB);
 
         final List<Subscription> expectedSubscriptions = testSubscriptions.stream()
-                .filter(sub -> "myapp1".equals(sub.getOwningApplication()))
                 .sorted(SUBSCRIPTION_CREATION_DATE_DESC_COMPARATOR)
                 .collect(toList());
+        expectedSubscriptions.subList(0, 2).clear();
```

Comment:
In this method I test pagination with offset=2 and limit=3.
So in expected list I first remove first two elements (because this is what offset=2 will do) and then I remove the tail of the list, keeping only 3 elements (because the limit=3). And in that way I get a list which I expect to get from repository method.